src/conn/conn_dhandle.c

```diff
@@ -134,7 +134,17 @@ __conn_dhandle_mark_dead(WT_SESSION_IMPL *session)
 	 * eviction and drain any blocks already queued.
 	 */
 	WT_RET(__wt_evict_file_exclusive_on(session, &evict_reset));
+
+	/* The handle is dead. */
 	F_SET(session->dhandle, WT_DHANDLE_DEAD);
+
+	/*
+	 * Reset the tree's eviction priority, and the tree is evictable by
+	 * definition.
+	 */
+	__wt_evict_priority_clear(session);
+	F_CLR(S2BT(session), WT_BTREE_NO_EVICTION);
+
 	if (evict_reset)
 		__wt_evict_file_exclusive_off(session);
```

Comment:
I don't follow all of the above, but maybe we just make the call to `__wt_evict_file_exclusive_off` here unconditional on `evict_reset`, along with a comment.  That seems better than clearing `NO_EVICTION` a couple of lines above, then conditionally doing it again in this call.